### PR TITLE
Add IT's for central validate-messages

### PR DESCRIPTION
**Pull request by prbans**
Issue: modifying validate/monitor code is fragile

Fix: add IT's, beef up IT's

Specifics:

* Added device client to dev_requirements so that test code can "mock" a device
* Added better "scenario" for monitor-events
* Added tests for validate-messages

Please wait for #212 to go in before this gets merged

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.

### CODE REVIEW

**Comment by prbans** [comment]
image

**Comment by prbans** [comment]
image

File: azext_iot/tests/helpers.py [hunk 1]
```diff
@@ -8,10 +8,46 @@
...
 
+GLOBAL_PROVISIONING_HOST = "global.azure-devices-provisioning.net"
+
 
 def load_json(filename):
     os.chdir(os.path.dirname(os.path.abspath(getsourcefile(lambda: 0))))
     return json.loads(read_file_content(filename))
+
+
+def dps_connect_device(device_id: str, credentials: dict) -> IoTHubDeviceClient:
```
Comment (digimaun) [inline azext_iot/tests/helpers.py:23]: Powerful helper overall in combination with device client. I think we'll definitely be interested in leveraging the device SDK more with device simulation capabilities.

File: azext_iot/tests/test_iot_central_int.py [hunk 2]
```diff
@@ -91,24 +93,138 @@ def test_central_device_twin_show_success(self):
...
+        )
+        device_client.send_message(msg)
+        expected_messages.append(strings.invalid_json())
+
+        # Validate the messages
+        output = self._get_validate_messages_output(
+            device_id, enqueued_time, max_messages=len(expected_messages)
+        )
+
+        self._delete_device(device_id)
```
Comment (digimaun) [inline azext_iot/tests/test_iot_central_int.py:213]: nit/future iteration thing: Consider looking into usage of fixtures and/or usage of xunit style setup & teardown to make it easier and faster to setup and clean up after a test.

File: pytest.ini.example [hunk 3]
```diff
@@ -23,8 +23,6 @@ env =
     azext_pnp_repository=
     azext_pnp_cs=
     azext_iot_central_app_id=
-    azext_iot_central_device_id=
```
Comment (digimaun) [inline pytest.ini.example:26]: Great, happy to see tests being easier to run. Can you please also remove the the relevant env vars from the [set-testenv-sentinal](https://github.com/Azure/azure-iot-cli-extension/blob/dev/.azure-devops/templates/set-testenv-sentinel.yml#L19) CI template?

File: azext_iot/central/_help.py [hunk 4]
```diff
@@ -107,6 +107,20 @@ def _load_central_devices_help():
             --device-id {deviceid}
     """
 
+    helps[
+        "iot central app device get-credentials"
```
Comment (digimaun) [inline azext_iot/central/_help.py:111]: For commands that provide say 'CRUDL' for a resource, as much as possible we want to stick with standard CLI verbs i.e. Create/Show/Update/Delete/List...for one off's there is more flexibility but why not change this command to `show-credentials`?

All that being said have a related comment on this command in a different file.

File: azext_iot/central/command_map.py [hunk 5]
```diff
@@ -51,6 +51,7 @@ def load_central_commands(self, _):
         cmd_group.command(
             "registration-summary", "registration_summary",
         )
+        cmd_group.command("get-credentials", "get_credentials")
```
Comment (digimaun) [inline azext_iot/central/command_map.py:54]: Is this command useful for IoT Central users? Clearly there is value for tests, but want to make sure this makes sense to expose to end users in the overall story for Central.

File: azext_iot/tests/test_iot_central_int.py (hunk 2 above)
Comment (prbans) [reply azext_iot/tests/test_iot_central_int.py:213]: Sounds good, I'll keep that in mind for next work items when they come down the pipe

File: azext_iot/central/command_map.py (hunk 5 above)
Comment (prbans) [reply azext_iot/central/command_map.py:54]: Lets chat about this one offline

**Comment by prbans** [comment]
image

**Pull request merged**